pkg/util/watcher/watch.go

```diff
@@ -0,0 +1,99 @@
+/*
```

Comment:
I am not sure if `pkg/util` is the best place for this? This definitely doesn't feel generic enough to be `util`. Thoughts @LiliC @brancz ?

Reply:
hmm any suggestions where it should live?